scripts/lib/CIME/XML/component.py

```diff
@@ -117,12 +122,145 @@ def _get_value_match(self, node, attributes=None, exact_match=False):
 
     #pylint: disable=arguments-differ
     def get_description(self, compsetname):
+        if self.get_version() == 3.0:
+            return self._get_description_v3(compsetname, comp_class=self._comp_class)
+        else:
+            return self._get_description_v2(compsetname)
+
+    def get_forcing_description(self, compsetname):
+        if self.get_version() == 3.0:
+            return self._get_description_v3(compsetname, comp_class='forcing')
+        else:
+            return ""
+
+
+    def _get_description_v3(self, compsetname, comp_class=None):
+        """
+        version 3 of the config_component.xml file has the description section at the top of the file
+        the description field has one attribute 'modifier_mode' which has allowed values
+        '*' 0 or more modifiers (default)
+        '1' exactly 1 modifier
+        '?' 0 or 1 modifiers
+        '+' 1 or more modifiers
+
+        modifiers are fields in the component section of the compsetname following the % symbol.
+
+        The desc field can have an attribute which is the component class ('cpl', 'atm', 'lnd' etc)
+        or it can have an attribute 'option' which provides descriptions of each optional modifier
+        or (in the config_component_{model}.xml in the driver only) it can have the attribute 'forcing'
+
+        component descriptions are matched to the compsetname using a set method
+        """
+        comp_class = comp_class.lower()
```

Comment:
`comp_class` defaults to `None`. Shouldn't there be a check here (like move the expect on line 159 before 154)?